openedx/openedx-platform
# [FC-0009] Paste Components (OLX) into any Unit in Studio

## Description

This continues the work from #31904 and implements ticket https://github.com/openedx/modular-learning/issues/12

Affects: course authors [for now, only those who will be participating in the beta test]

Screenshot of the "Paste Component" button:

![Screenshot 2023-03-20 at 10 51 15 AM](https://user-images.githubusercontent.com/945577/226425148-6546dea2-61a6-4c41-9b00-1758f830dd4a.png)

Demo video: coming 

## Supporting information

See tickets linked above.

## Testing instructions

1. Go to (studio)/admin/waffle/flag/ and enable `contentstore.enable_copy_paste_feature` for Everyone, if you haven't already.
2. Open two browser tabs so you can see both at the same time. Ideally, they're in different courses.
3. Use the new Actions menu to select **Copy**. You should see an animation and the "Paste Component" button should instantly appear in any open browser tabs.
4. Try pasting into the same course and different course.
5. Refresh the page and ensure the "Paste Component" button still appears
6. Try various combinations of copying and pasting
    - e.g. try copy, then delete the source block, then paste. It should still work!
    - e.g. try within the same course, multiple pastes into the same unit (should get unique IDs), into different courses, etc.

Note: handling of static assets, intra-course links, pythonlib.zip, content groups, and anything along those lines is out of scope for now. Pasting within a course should always work fine. Pasting between courses may have issues if the component uses one or more of these features.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Private ref: MNG-3605
